src/common/k8s-api/endpoints/cluster.api.injectable.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import assert from "assert";
+import { createStoresAndApisInjectionToken } from "../create-stores-apis.token";
+import { ClusterApi } from "./cluster.api";
+
+const clusterApiInjectable = getInjectable({
+  id: "cluster-api",
+  instantiate: (di) => {
+    assert(di.inject(createStoresAndApisInjectionToken), "clusterApi is only available in certain environments");
```

Comment:
Here `di.inject()` is used primarily to trigger something in `instantiate()`, instead of using the returned instance. This is misleading, as something called instantiate exists to create instances. Also the result of injection, `createStoresAndApis` cannot be used for what the name implies, and instead of a function there's a boolean for unclear intent.

Reply:
This is a guard statement, is there a better way to do this since you keep mentioning it. Even though the fact that it is guard statement seems obvious.

Reply:
This thread was resolved without resolution. Manors, gentlemen.

How about what is proposed [here](https://github.com/lensapp/lens/pull/5195#discussion_r872175772)? That would at least mean we could get rid of the 38 duplications.